Runtime error due to nil pointer dereference on TargetRef fields missing

When a ScaledObject resource is created but it's missing both the `scaledTargetRef` and the `jobTargetRef` fields, the controller assumes its `ScaleType` is a "job" but then a runtime error happens due to a nil pointer dereference.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ef8f0a]

goroutine 468 [running]:
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).resolveJobEnv(0xc0008ce660, 0xc000347ba0, 0x0, 0x0, 0x0)
	/home/ppatiern/github/keda/pkg/handler/scale_jobs.go:99 +0x5a
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).getJobScalers(0xc0008ce660, 0xc000347ba0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ppatiern/github/keda/pkg/handler/scale_handler.go:224 +0xaf
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).handleScaleJob(0xc0008ce660, 0x2d88a40, 0xc00059f700, 0xc000347ba0)
	/home/ppatiern/github/keda/pkg/handler/scale_loop.go:53 +0xe5
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).handleScale(0xc0008ce660, 0x2d88a40, 0xc00059f700, 0xc000347ba0)
	/home/ppatiern/github/keda/pkg/handler/scale_loop.go:42 +0xbe
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).HandleScaleLoop(0xc0008ce660, 0x2d88a40, 0xc00059f700, 0xc000347ba0)
	/home/ppatiern/github/keda/pkg/handler/scale_loop.go:14 +0x317
created by github.com/kedacore/keda/pkg/controller/scaledobject.(*ReconcileScaledObject).startScaleLoop

/home/ppatiern/github/keda/pkg/controller/scaledobject/scaledobject_controller.go:318 +0x5d3
```

It happens in the `resolveJobEnv` method.
I would fix it just doing a check in the caller method `getJobScalers` that the `jobTargetRef` is not nil and returning an error eventually. Maybe we should do the same in the `GetDeploymentScalers` even if it should not happen.

--- a/pkg/controller/scaledobject/scaledobject_controller.go
+++ b/pkg/controller/scaledobject/scaledobject_controller.go
@@ -148,13 +148,25 @@ func (r *ReconcileScaledObject) Reconcile(request reconcile.Request) (reconcile.
 	}
 
 	reqLogger.V(1).Info("Detecting ScaleType from ScaledObject")
-	if scaledObject.Spec.ScaleTargetRef == nil || scaledObject.Spec.ScaleTargetRef.DeploymentName == "" {
+	var errMsg string
+	if scaledObject.Spec.ScaleTargetRef != nil {
+		if scaledObject.Spec.JobTargetRef == nil {
+			reqLogger.Info("Detected ScaleType = Deployment")
+			return r.reconcileDeploymentType(reqLogger, scaledObject)
+		}
+		errMsg = "Both ScaledObject.Spec.ScaleTargetRef and ScaledObject.Spec.JobTargetRef cannot be set at the same time"
+	} else if scaledObject.Spec.JobTargetRef != nil {
 		reqLogger.Info("Detected ScaleType = Job")
 		return r.reconcileJobType(reqLogger, scaledObject)
 	} else {
-		reqLogger.Info("Detected ScaleType = Deployment")
-		return r.reconcileDeploymentType(reqLogger, scaledObject)
+		errMsg = "ScaledObject.Spec.ScaleTargetRef or ScaledObject.Spec.JobTargetRef is not set"
 	}
+	if errMsg == "" {
+		errMsg = "Unknown error while detecting ScaleType"
+	}
+	err = fmt.Errorf(errMsg)
+	reqLogger.Error(err, "Failed to detect ScaleType")
+	return reconcile.Result{}, err
 }
 
 // reconcileJobType implemets reconciler logic for K8s Jobs based ScaleObject

